cfme/automate/import_export.py

```diff
@@ -1,70 +1,112 @@
 # -*- coding: utf-8 -*-
+import attr
+from navmazing import NavigateToAttribute
+from widgetastic.widget import Checkbox
+from widgetastic.widget import FileInput
+from widgetastic.widget import Text
+from widgetastic.widget import View
 from widgetastic_patternfly import BootstrapSelect
 from widgetastic_patternfly import Button
+from widgetastic_patternfly import FlashMessages
+from widgetastic_patternfly import Input
 
-from cfme.base.ui import AutomateImportExportBaseView
-from cfme.base.ui import AutomateImportExportView
-from cfme.utils.appliance import Navigatable
+from cfme.base.login import BaseLoggedInPage
+from cfme.modeling.base import BaseCollection
+from cfme.modeling.base import BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep
 from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.appliance.implementations.ui import navigator
 
 
-class GitImportSelectorView(AutomateImportExportBaseView):
-    type = BootstrapSelect('branch_or_tag')
-    branch = BootstrapSelect(locator='.//div[contains(@class, "bootstrap-select git-branches")]')
-    tag = BootstrapSelect(locator='.//div[contains(@class, "bootstrap-select git-tags")]')
+class AutomateImportExportBaseView(BaseLoggedInPage):
+    flash = FlashMessages("div.import-flash-message")
+    title = Text('.//div[@id="main-content"]//h1')
 
-    submit = Button('Submit')
-    cancel = Button('Cancel')
+    @property
+    def in_import_export(self):
+        return (
+            self.logged_in_as_current_user
+            and self.navigation.currently_selected == ["Automation", "Automate", "Import / Export"]
+            and self.title.text == "Import / Export"
+        )
 
     @property
     def is_displayed(self):
-        return self.in_import_export and self.type.is_displayed
+        return self.in_import_export
 
 
-class AutomateGitRepository(Navigatable):
-    """Represents an Automate git repository. This entity is not represented in UI as it is, but
-    only in database. But by representing it it makes the code changes for domain much simpler.
+class AutomateImportExportView(AutomateImportExportBaseView):
+    @View.nested
+    class import_file(View):  # noqa
+        file = FileInput(id="upload_file")
+        upload = Button(id="upload-datastore-import")
 
-    """
-    def __init__(self, url=None, username=None, password=None, verify_ssl=None, appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
-        self.url = url
-        self.username = username
-        self.password = password
-        self.verify_ssl = verify_ssl
-        self.domain = None
+    @View.nested
+    class import_git(View):  # noqa
+        ROOT = './/form[@id="retrieve-git-datastore-form"]'
```

Comment:
Question : What is this for ?

Reply:
to set root of `View`. we can set `ROOT` of widget or view as per `wt`. btw, i just took this from old implementation.

Reply:
When the widgets within this view are resolved, they will use this ROOT locator to establish the portion of the page DOM to search in.

Its not necessary here, but results in more explicit widget lookup - image you had a page with each section containing an `Add` button. You could setup nested views for each section, using ROOT like this, to make sure the `Add` button that's found via xpath will be the one under the ROOT locator.